services/spar/src/Spar/Data.hs

```diff
@@ -419,6 +429,47 @@ deleteIdPRawMetadata idp = retry x5 . write del $ params Quorum (Identity idp)
     del :: PrepQuery W (Identity SAML.IdPId) ()
     del = "DELETE FROM idp_raw_metadata WHERE id = ?"
 
+----------------------------------------------------------------------
+-- SSO settings
+
+-- It's important to maintain two invariants:
+-- 1) whenever there is a default code, it must also exist in the idp table
+-- 2) there can always only be one default SSO code selected
+
+getDefaultSSOCode
+  :: (HasCallStack, MonadClient m)
+  => m (Maybe SAML.IdPId)
+getDefaultSSOCode = fmap runIdentity . minimumMay <$>
+  (retry x1 . query sel $ params Quorum ())
+  where
+    sel :: PrepQuery R () (Identity SAML.IdPId)
+    sel = "SELECT idp FROM default_idp"
```

Comment:
I am still thinking about whether this is the schema we want. This query works, but based on what I read about Cassandra so far, it doesn't know which partition the one row is on, so it has to ask all nodes whether they have it. This would not be the case with a primary key with just one possible value.